packages/nx/src/native/pseudo_terminal/process_killer/windows.rs

```diff
@@ -0,0 +1,42 @@
+use winapi::shared::minwindef::DWORD;
+use winapi::shared::ntdef::HANDLE;
+use winapi::um::processthreadsapi::{OpenProcess, TerminateProcess};
+use winapi::um::winnt::{PROCESS_QUERY_INFORMATION, PROCESS_TERMINATE};
+
+pub struct ProcessKiller {
+    pid: i32,
+}
+
+impl ProcessKiller {
+    pub fn new(pid: i32) -> Self {
+        Self { pid }
+    }
+
+    pub fn kill(&self, signal: Option<String>) -> anyhow::Result<()> {
+        let pc = WinProcess::open(pid as DWORD).expect("!open");
+        pc.kill()
+            .map_err(|e| anyhow::anyhow!("Failed to kill process {}: {}", self.pid, e))?;
+    }
+}
+
+struct WinProcess(HANDLE);
+impl WinProcess {
+    fn open(pid: DWORD) -> Result<Process, String> {
+        // https://msdn.microsoft.com/en-us/library/windows/desktop/ms684320%28v=vs.85%29.aspx
+        let pc = unsafe { OpenProcess(PROCESS_QUERY_INFORMATION | PROCESS_TERMINATE, 0, pid) };
```

Comment:
Is this what portable_pty does under the hood? Why is this unsafe?

Reply:
`OpenProcess` is marked unsafe, so this has to be unsafe. Its just one of those `winapi` things unfortunately

Reply:
I don't think portable pty has to use OpenProcess... because they are the one that spawned the command. But winapi has OpenProcess marked as `unsafe`, so calling it has to be `unsafe`.

We don't have the windows instance of the process and have to fetch it by pid, and this is the winapi provided way of doing so.